### PR TITLE
US028: Add Finance Section For Zone Cards

### PR DESCRIPTION
**Pull request by abdullokhtolibjonov**

### CODE REVIEW

File: Parking-Zone/Areas/Admin/Controllers/ReservationsController.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    public class ReservationsController : Controller
+    {
+        private readonly IReservationService _reservationService;
+
+        public ReservationsController(IReservationService reservationService)
+        {
+            _reservationService = reservationService;
+        }
+
+        public IActionResult Index(string period)
```
Comment (DilshodMQ) [inline Parking-Zone/Areas/Admin/Controllers/ReservationsController.cs]: Add this Action into ParkingZoneController

File: Parking-Zone/Areas/Admin/Views/ParkingZone/Index.cshtml [hunk 2]
```diff
@@ -28,16 +28,59 @@
...
+                <p class="card-text" style="color: #666; font-family: 'Arial', sans-serif;">Free Slots: @item.FreeSlots</p>
+                <a asp-controller="ParkingSlot" asp-action="Index" asp-route-zoneId="@item.Id" class="btn btn-primary" style="font-family: 'Quicksand'">Parking Slots <i class="bi bi-list"></i></a>
+                <a asp-controller="ParkingZone" asp-action="CurrentCars" asp-route-zoneId="@item.Id" class="btn btn-primary" style="font-family: 'Quicksand'; background-color: #8A2BE2; border-color: #8A2BE2; color: #fff;">Current Cars <i class="bi bi-car-front-fill"></i></a>
+            </div>
+
+            <div>
+                <h5 style="margin-right: 400px;">Finance summary</h5>
+                <div class="d-flex">
+                    <h5 style="margin-right: 10px;">Period For</h5>
+                    <select id="periodSelect" class="form-select" aria-label="Default select example" style="width: 165px;">
```
Comment (DilshodMQ) [inline Parking-Zone/Areas/Admin/Views/ParkingZone/Index.cshtml]: use enum type in this select, and use html helper to get values of enum

File: Parking-Zone/Services/ReservationService.cs [hunk 3]
```diff
@@ -26,5 +27,31 @@ public void Prolong(Reservation reservation, int extraHours)
             reservation.Duration += extraHours;
             Update(reservation);
         }
+
+        public ReservationHoursSummaryVM GetStandardAndBusinessHoursByPeriod(string period)
```
Comment (DilshodMQ) [inline Parking-Zone/Services/ReservationService.cs]: It is not VM

**Review by DilshodMQ — CHANGES_REQUESTED** [review]

File: Tests/Services/ReservationServiceTests.cs [hunk 4]
```diff
@@ -52,5 +103,65 @@ public void GivenAppUserId_WhenGetByAppUserIdIsCalled_ThenAllReservationsReturne
             mockReservationRepository.Verify(repo => repo.GetAll(), Times.Once);
         }
         #endregion
+
+        #region GetStandardAndBusinessHoursByPeriod
+        [Fact]
+        public void GivenReservationsAndAllTimePeriod_WhenGetStandardAndBusinessHoursByPeriodIsCalledThenReturnedAllReservationSummaryHours()
+        {
```
Comment (DilshodMQ) [inline Tests/Services/ReservationServiceTests.cs]: add tests for other cases

**Review by DilshodMQ — CHANGES_REQUESTED** [review]

**Review by DilshodMQ — APPROVED** [review]

**Comment by abdullokhtolibjonov** [comment]
[US028 (3): Modify Zone card: Also display Finance block with ability to filter by period to see number of hours booked for each category. Use jQuery ajax for filtering.](https://trello.com/c/gUxJHusM/37-us028-3-modify-zone-card-also-display-finance-block-with-ability-to-filter-by-period-to-see-number-of-hours-booked-for-each-cate)

File: Parking-Zone/Services/ParkingZoneService.cs [hunk 5]
```diff
@@ -24,5 +24,29 @@ public List<string> GetCurrentCarsPlateNumbersByZone(ParkingZone zone)
...
+
+            zoneFinanceData.CategoryHours = slots
+                .GroupBy(s => s.Category)
+                .ToDictionary(
+                    g => g.Key,
+                    g => g.SelectMany(r => r.Reservations
+                                                         .Where(r => r.StartTime >= startInclusive && r.StartTime < endExclusive))
+                                                         .Sum(r => r.Duration));
+
+            //foreach (var reservation in reservationsForPeriod)
```
Comment (napster1991) [inline Parking-Zone/Services/ParkingZoneService.cs]: Remove comment

**Review by napster1991 — APPROVED** [review]

**Pull request merged**